ztoc/ztoc_marshaler.go

```diff
@@ -97,7 +97,8 @@ func flatbufToZtoc(flatbuffer []byte) (z *Ztoc, err error) {
 		dgst, _ := digest.Parse(string(compressionInfo.SpanDigests(i)))
 		ztoc.SpanDigests[i] = dgst
 	}
-	ztoc.Checkpoints = compressionInfo.CheckpointsBytes()
+	ztoc.Checkpoints = make([]byte, len(compressionInfo.CheckpointsBytes()))
+	copy(ztoc.Checkpoints, compressionInfo.CheckpointsBytes())
```

Comment:
Maybe it be helpful to add a comment here that `CheckpointsBytes` returns a slice so need to manually copy for garbage collection to reclaim the flatbuffer memory.

Also any thoughts on validation? Are there any unit tests or integration tests that should be included with this change?

Reply:
It might be tough. Instinctively I would like to compare pointer addresses, but it would be tough to do so since directly referencing pointers is [considered unsafe](https://go101.org/article/unsafe.html).

It also is a somewhat weird thing to test, since it's specifically testing one field in one struct in one part of the code that's only done in the beginning of the code and never again. I think it would be cool to do a general test to try and make sure we're not holding pointers that we shouldn't be, but that seems like a much larger scope thing and probably needs to be implemented object by object or something like that.

I think it might be ok to not test this. Feel free to resolve if you agree, otherwise open to more ideas.